cupyx/scipy/sparse/_compressed.py

```diff
@@ -416,17 +416,24 @@ def _major_index_fancy(self, idx):
     def _minor_index_fancy(self, idx):
         """Index along the minor axis where idx is an array of ints.
         """
-        M, _ = self._swap(*self.shape)
+        M, N_org = self._swap(*self.shape)
         N = idx.size
         new_shape = self._swap(M, N)
         if self.nnz == 0 or N == 0:
             return self.__class__(new_shape, dtype=self.dtype)
-
         if idx.size * M < self.nnz:
             # TODO (asi1024): Implement faster algorithm.
             pass
 
-        return self._tocsx()._major_index_fancy(idx)._tocsx()
+        idx_is_sorted = cupy.all(idx[:-1] <= idx[1:])
+        if not self.has_sorted_indices or idx_is_sorted:
```

Comment:
`idx_is_sorted` is a `cupy.ndarray`, so it requires DtoH synchronization. Aligning with CuPy's policy, we provide features as possible while minimizing the amount of DtoH synchronization.
How about sorting `idx` without checking it is already sorted?

Reply:
I feel like sorting the index would create problems for the user. Because some people relay on sub-setting to sort there array.

Reply:
How about copying `idx`?

Reply:
I'm sorry, but I don't understand why copying would help with the issue. I added this:
```
        idx_is_sorted = cupy.all(idx[:-1] <= idx[1:])
        if not self.has_sorted_indices or idx_is_sorted:
```
because I want the user to get sorted `X.indices` if `X.has_sorted_indices`. If we don't care about that, we could leave this out and just run with the shortcut in every case. However, calling `X.sort_indices()` is more expensive than the old solution (which still has one DtoH that I save).

Reply:
Hmm, I would prefer not to perform additional computation on the GPU to decide whether to use a shortcut path or not. Could you modify to allow shortcuts only to the extent that they can be branched by simple calculations on the host side?